### ISSUE TITLE
S2857 has a false positive

### ISSUE THREAD

**Issue by Sergey1209** [SonarSource/sonar-dotnet#4778]
### Description

Rule: S2857
Evaluates C# "string.Join()"  as SQL statement

### Repro steps

Use "string.Join()" in C# sample

### Expected behavior

C# operators should not be evaluated as SQL

### Actual behavior

C# "string.Join()" is evaluated as SQL statement

### Known workarounds

None

### Related information

* C#7
* Visual Studio 2019
* MSBuild 15
* SonarScanner for .NET version 4.21
* Windows 10

**Comment by pavel-mikula-sonarsource** [comment]
Hi @Sergey1209 ,

can you please provide a small code fragment to reproduce the scenario?

**Comment by Sergey1209** [comment]
Sure: 

throw new Exception($"{string.Join($"{Environment.NewLine}", someArray)} {Environment.NewLine}");

**Comment by pavel-mikula-sonarsource** [comment]
Hi, I can't reproduce it. I tried the case below, but I'm not getting the issue on the `throw` line.

What analyzer version are you using? If you don't know the analyzer version, can you tell us version of SonarQube you're using?
```
using System;
using System.Data.SqlClient;

namespace Something
{
    class Repro
    {
        public void Go(string[] someArray)
        {
            var bulkInsert = "BULK INSERT AdventureWorks2012.Sales.SalesOrderDetail" +"FROM 'neworders.txt';"; // Raised here, as expected
            throw new Exception($"{string.Join($"{Environment.NewLine}", someArray)} {Environment.NewLine}");
        }
    }
}

```
